Tradehouse

Deatomizeer opened the pull request.

Deatomizeer committed:
backend: added a tradehouse table in database and a trade offer's model

Deatomizeer committed:
backend: implemented partial tradehouse functionality

Added methods for creating offers and retrieving the existing ones.

KamilMatejuk committed:
client: added resource selection

Deatomizeer committed:
database: added a procedure allowing players to trade resources

Deatomizeer committed:
backend: added a method for accepting trade offers and slightly adjusted their model

Now the resource type is the string enum type instead of the number enum one.

jerry-sky committed:
backend: added tradehouse routes

Deatomizeer committed:
backend: added more error handling related to tradehouse feature

A trade cannot be accepted unless both parties have sufficient funds to complete it.

KamilMatejuk committed:
client: added methods for tradehouse

KamilMatejuk committed:
client: create offer returns new id

KamilMatejuk commented:
Could backend team make it so creating offer request returns an id of the newly created offer. This way we wouldn't have to reload all offers each time a new one is created.
Note: the last commit expects it to return number, so right now creates an error. Commit `8b6ab70` works ok.

Deatomizeer committed:
backend: minor change to trade creation method

CreateTradeOffer() now returns the newly created offer's ID instead of nothing.

KamilMatejuk committed:
client: added reloading user data after action

KamilMatejuk committed:
client: fixed removing resources on offer creation

KamilMatejuk committed:
client: hotfix city error

Deatomizeer committed:
game design: added values related to buildings

All buildings now have their upgrade cost and hourly resource output specified in proper JSON files.
Full credit for the commit goes to Necr0o0.

Deatomizeer committed:
Merge branch 'tradehouse' of https://github.com/jerry-sky/cyber-city-sim into tradehouse

KamilMatejuk committed:
client: final style touches

KamilMatejuk committed:
client: removed buttons to private messages
